alg/teca_table_reduce.cxx

```diff
@@ -64,24 +64,21 @@ p_teca_dataset teca_table_reduce::reduce(
 
     p_teca_table output_table;
 
-    bool left = left_table && *left_table;
-    bool right = right_table && *right_table;
-
-    if (left && right)
+    if (left_table && right_table)
```

Comment:
previously it looks like you were checking for the existence of the `left_table` pointer and a valid value for it; now it just looks like you're checking for the pointer existence.  If I'm understanding that right, was the previous code redundant in some way?  Or has the `teca_table` changed in some way that it has more internal safety checks?  (And this isn't something that needs to be addressed prior to merge: it's more a curiosity.)

Reply:
good question.  The reason for the change is that in the new streaming mode one of the tables can be empty, and it is a case that will now occur regularly.
```
bool right = right_table && *right_table;
```
In the statement `right_table` evaluates true if the shared pointer is not null. `*right_table` evaluates true if the table it points to is not empty. I implemented an automatic conversion to bool. Here is the code for the latter
```
    explicit operator bool() const noexcept
    { return !this->empty(); }
```
These type of conversions while convenient are one of the more esoteric features of C++,  can be tricky to read (especially when mixed with smart pointers), and have some gotchas that can cause problems if one is not careful with them(the explicit helps). For better or worse I ignored all that and implemented it. We could potentially remove such conversions from the code base to improve readability and make it harder to use incorrectly.

Reply:
ah, got it, that makes a lot of sense.  No, I don't think there's any need to change what's there.  Thanks for explaining.

Reply:
Now that I'm thinking about this more, the case that comes up in streaming mode is where the right table is a nullptr. I don't recall why I took the check for the empty table out. I think it is an edge case that doesn't come up too often. However, I'll do my best to double check the code to make sure there are no lurking issues here.

Reply:
I added the check for empty tables back in.